rdt-app/src/main/java/io/ona/rdt_app/widget/CustomRDTCaptureFactory.java

```diff
@@ -68,4 +83,45 @@ protected void launchRDTCaptureActivity() {
             new LaunchRDTCameraTask().execute(intent);
         }
     }
+
+    private OnActivityResultListener createOnActivityResultListener() {
+
+        OnActivityResultListener resultListener =  new OnActivityResultListener() {
+
+            @Override
+            public void onActivityResult(int requestCode, int resultCode, Intent data) {
+                hideProgressDialog();
+                final JsonApi jsonApi = (JsonApi) context;
+                if (requestCode == JsonFormConstants.RDT_CAPTURE_CODE && resultCode == RESULT_OK) {
+                    if (data != null) {
+                        try {
+                            String[] imgIDAndTimeStamp = data.getExtras().getString(SAVED_IMAGE_FILE_PATH).split(",");
+                            jsonApi.writeValue("step4", RDT_CAPTURE, imgIDAndTimeStamp[0], "", "", "", false);
+                            jsonApi.writeValue("step5", TIME_IMG_SAVED, imgIDAndTimeStamp[1], "", "", "", false);
```

Comment:
Is this a good idea changing other keys from a factory.

Why not change the value of this component `jsonObject`.

You can also store additional params and use calculations on other steps to retrieve this value